### PR TITLE
Check for duplicate turn numbers in store invariant check

### LINKED ISSUE #1554: Store should only store one state per turn number
https://github.com/statechannels/monorepo/issues/1555 asserts that the store should only support one state per turn number, within a given channel.

The store has no control over what state a peer sends for turn `n`. If one of my workflows subsequently want to support a different state at turn `n`, what should the store do?

Right now, some store logic depends (might depend?) on the fact that there's only one state per turn number.

See https://github.com/statechannels/monorepo/issues/1456#issuecomment-615520101

### PR DESCRIPTION
**Pull request by lalexgap**
Fixes #1554 by throwing an error when we have multiple states with the same turn number.

This would of caught #1729

### CODE REVIEW

**Comment by snario** [comment]
Adding a state with the same store number is a legitimate use case though, for example in doing a unidirectional payment channel where the receiver can conclude at any time.

**Comment by andrewgordstewart** [comment]
> Adding a state with the same store number is a legitimate use case though,

More than that, if we are going to _store_ every state that a peer sends us, this would prevent us from adding our own states with those turn numbers.

> for example in doing a unidirectional payment channel where the receiver can conclude at any time.

I don't understand this example. Can you expand?

**Comment by snario** [comment]
> I don't understand this example. Can you expand?

- Sender wants to send at most 5 coins to Receiver in increments of 1
- Sender signs `{ turnNum: 0, isFinal: true, outcome: [5, 0] }`
- Sender signs `{ turnNum: k, isFinal: true, outcome: [5 - k, k] }`
- Sender goes offline for some `k` (e.g., 3)
- Receiver wants on-chain cash _now_ (no timeout)
- Receiver concludes with `{ turnNum: k + 1, isFinal: true, outcome: [5 -j, j] }` for max `k`

**Comment by andrewgordstewart** [comment]
In this particular case, sender can sign `{ turnNum: k, isFinal: true, outcome: [5-k, k] }`

**Comment by lalexgap** [comment]
> Adding a state with the same store number is a legitimate use case though, for example in doing a unidirectional payment channel where the receiver can conclude at any time.

Currently our `store`/`api` doesn't properly support the case of multiple states per a turn number. I think we should support this eventually but I think there's enough details there that we shouldn't try and do it now.

File: packages/xstate-wallet/src/store/channel-store-entry.ts [hunk 1]
```diff
@@ -257,6 +257,11 @@ export class ChannelStoreEntry {
     const {signedStates} = this;
     const turnNums = _.map(signedStates, s => parseInt(s.turnNum.toHexString(), 16));
 
+    const duplicateTurnNums = turnNums.some((t, i) => turnNums.indexOf(t) != i);
+    if (duplicateTurnNums) {
+      logger.error({signedStates: _.map(signedStates, s => s.turnNum.toHexString())});
```
Comment (andrewgordstewart) [inline packages/xstate-wallet/src/store/channel-store-entry.ts:262]:
```suggestion
      logger.error(
        {signedStates: _.map(signedStates, s => s.turnNum.toHexString())},
        Errors.duplicateTurnNums
      );
```

Nice to have a string to filter by in the browser

**Pull request merged**